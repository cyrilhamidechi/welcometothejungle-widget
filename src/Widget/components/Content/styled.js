import styled from "styled-components";

import { colors } from "./../../../config";

export const Content = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
`;

export const Cover = styled(Content)`
  background-image: ${({ caption }) => caption && `url(${caption})`};
  background-size: cover;
  background-position: center;
  color: ${colors.white};
  transition: opacity 300ms ease;
`;

// Create wrapper with ratio 16/9
export const Wrapper = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  color: ${colors.white};
  overflow: hidden;

  &:hover {
    ${Cover} {
      opacity: 0.8;
    }
  }
`;

export const Button = styled.div`
  transform: scale(1);
  transition: transform 300ms ease;
`;

/* PICTURE COMPONENT */
export const Picture = styled.div`
  display: flex;
  opacity: 0;
  align-items: center;
  justify-content: center;
  height: 100%;

  &:hover {
    opacity: 1;

    ${Button} {
      transform: scale(1.5);
    }
  }
`;

/* VIDEO COMPONENT */
export const Video = styled.div`
  display: flex;
  justify-content: flex-end;
  flex-direction: column;
  height: 100%;

  &:hover {
    ${Button} {
      transform: scale(1.2);
    }
  }
`;

export const VideoItem = styled.div`
  display: flex;
  background-image: linear-gradient(transparent, rgba(0, 0, 0, 0.6));
  padding: 20px 10px 10px;
  font-size: 14px;
`;

export const VideoTitle = styled.div`
  margin-left: 10px;
`;

/* QUOTE COMPONENT */
export const QuoteIcon = styled.div`
  transition: margin-bottom 250ms ease;
`;

export const Quote = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  height: 100%;
  padding: 20px;
  box-sizing: border-box;
  font-size: 18px;

  &:hover {
    ${QuoteIcon} {
      margin-bottom: 5px;
    }
  }
`;
