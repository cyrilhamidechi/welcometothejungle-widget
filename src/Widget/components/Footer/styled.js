import styled from "styled-components";

import { colors } from "./../../../config";

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 0 10px;
  width: 100%;
  height: 50px;
  background: ${colors.black};
`;

export const PoweredBy = styled.a`
  display: flex;
  align-items: center;
`;
