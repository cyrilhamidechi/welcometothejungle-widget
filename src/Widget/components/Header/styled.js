import styled from "styled-components";

import { colors } from "./../../../config";

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 0 10px;
  width: 100%;
  height: 50px;
  background: ${colors.green};
  font-weight: 700;
  box-sizing: border-box;
`;

export const Logo = styled.img`
  max-height: 40px;
  margin-right: 10px;
`;

export const Next = styled.div`
  margin-left: 5px;
`;
