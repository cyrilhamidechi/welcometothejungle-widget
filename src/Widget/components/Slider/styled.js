import styled from "styled-components";

import { colors } from "./../../../config";

export const Wrapper = styled.div`
  display: flex;
  height: 100%;
  overflow: hidden;
  background: ${colors.black};
  justify-content: space-between;
`;

export const Scroll = styled.div`
  transform: ${({ translate }) => `translate3d(-${translate}px, 0, 0)`};
  transition: transform 600ms cubic-bezier(0.65, 0.05, 0.36, 1);
`;

export const Slide = styled.div`
  width: ${({ width }) => `${width}px`};
  display: flex;
  flex-wrap: wrap;
  height: 100%;
`;

export const Content = styled.div`
  width: 100vw;

  @media (min-width: 640px) {
    width: ${({ columns }) => `calc(100vw / ${columns})`};
  }
`;
