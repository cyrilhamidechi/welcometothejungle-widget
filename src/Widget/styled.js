import styled from "styled-components";
import { colors } from "./../config";

export const Wrapper = styled.div`
  border-radius: 6px;
  font-family: "Roboto Condensed", sans-serif;
  color: ${colors.black};
`;
