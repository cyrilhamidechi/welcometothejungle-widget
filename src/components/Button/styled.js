import styled from "styled-components";

import { colors } from "../../config";
import { getBackgroundHover } from "./utils";

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: ${({ onlyIcon }) => onlyIcon && "30px"};
  height: 30px;
  padding: ${({ onlyIcon }) => (onlyIcon ? "0" : "0 20px")};
  border: 0;
  border-radius: ${({ onlyIcon }) => (onlyIcon ? "30px" : "4px")};
  background: ${({ color }) => colors[color]};
  color: ${colors.white};
  font-size: 14px;
  text-decoration: none;
  cursor: pointer;
  transition: background 300ms ease-in-out, opacity 300ms;

  &:hover:enabled {
    background: ${getBackgroundHover};
  }

  &:disabled {
    opacity: 0.3;
  }
`;

export const Text = styled.span`
  display: flex;
  align-items: center;
  margin-left: "10px";
`;
