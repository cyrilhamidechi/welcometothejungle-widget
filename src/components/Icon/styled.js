import styled, { css } from "styled-components";

import { color } from "./utils";

export const Wrapper = styled.span`
  display: inline-block;
  line-height: 1;
  svg {
    width: 100%;
    height: 100%;
  }

  path {
    transition: fill 300ms;
  }

  ${props =>
    props.size &&
    css`
      height: ${props.size}px;

      svg {
        width: ${props.size}px;
        height: ${props.size}px;
      }
    `};

  ${props =>
    props.color &&
    css`
      svg path {
        fill: ${color(props.color)}
    `};
`;
