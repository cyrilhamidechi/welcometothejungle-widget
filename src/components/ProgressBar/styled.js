import styled from "styled-components";

import { colors } from "./../../config";

export const Wrapper = styled.div`
  width: 100%;
  height: 3px;
  background: ${colors.greyLight};
`;

export const ProgressBar = styled.div`
  width: ${({ progress }) => `${progress}%`}
  height: 100%;
  background: ${colors.green}
  transition: width 300ms ease-in-out;
`;
