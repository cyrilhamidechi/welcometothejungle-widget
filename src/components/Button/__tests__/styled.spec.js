import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Wrapper } from '../styled'

describe('Button styled', () => {
  describe('Wrapper', () => {
    it('should render the button with a onlyIcon', () => {
      const tree = renderer
        .create(
          <Wrapper onlyIcon>
            <div>Click me</div>
          </Wrapper>
        )
        .toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('width', '30px')
      expect(tree).toHaveStyleRule('border-radius', '30px')
      expect(tree).toHaveStyleRule('padding', '0')
    })

    it('should render the button with a onlyIcon', () => {
      const tree = renderer
        .create(
          <Wrapper>
            <div>Click me</div>
          </Wrapper>
        )
        .toJSON()

      expect(tree).toMatchSnapshot()
      expect(tree).toHaveStyleRule('width', undefined)
      expect(tree).toHaveStyleRule('border-radius', '4px')
      expect(tree).toHaveStyleRule('padding', '0 20px')
    })
  })
})