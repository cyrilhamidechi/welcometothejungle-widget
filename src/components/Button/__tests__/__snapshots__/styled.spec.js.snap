// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button styled Wrapper should render the button with a onlyIcon 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  padding: 0;
  border: 0;
  border-radius: 30px;
  color: #fff;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background 300ms ease-in-out;
  transition: background 300ms ease-in-out;
}

.c0:hover {
  background: #37cbaa;
}

<div
  className="c0"
>
  <div>
    Click me
  </div>
</div>
`;

exports[`Button styled Wrapper should render the button with a onlyIcon 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 30px;
  padding: 0 20px;
  border: 0;
  border-radius: 4px;
  color: #fff;
  font-size: 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: background 300ms ease-in-out;
  transition: background 300ms ease-in-out;
}

.c0:hover {
  background: #37cbaa;
}

<div
  className="c0"
>
  <div>
    Click me
  </div>
</div>
`;
